homeassistant/components/sensor/dwd_weather_warnings.py

```diff
@@ -0,0 +1,242 @@
+    'advance_warning_level': ['Advance Warning Level',
+                              None, 'mdi:close-octagon-outline'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_REGION_NAME, default=DEFAULT_REGION_NAME): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(MONITORED_CONDITIONS)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the DWD-Weather-Warnings sensor."""
+    name = config.get(CONF_NAME)
+    region_name = config.get(CONF_REGION_NAME)
+
+    api = DwdWeatherWarningsAPI(region_name)
+
+    sensors = [DwdWeatherWarningsSensor(hass, api, name, condition)
+               for condition in config[CONF_MONITORED_CONDITIONS]]
+
+    add_devices(sensors, True)
+
+
+class DwdWeatherWarningsSensor(Entity):
+    """Representation of a DWD-Weather-Warnings sensor."""
+
+    def __init__(self, hass, api, name, variable):
+        """Initialize a DWD-Weather-Warnings sensor."""
+        self._hass = hass
+        self._api = api
+        self._name = name
+        self._var_id = variable
+
+        variable_info = MONITORED_CONDITIONS[variable]
+        self._var_name = variable_info[0]
+        self._var_units = variable_info[1]
+        self._var_icon = variable_info[2]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "{} {}".format(self._name, self._var_name)
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._var_icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._var_units
+
+    # pylint: disable=no-member
+    @property
+    def state(self):
+        """Return the state of the device."""
+        try:
+            return round(self._api.data[self._var_id], 2)
+        except TypeError:
+            return self._api.data[self._var_id]
+
+    # pylint: disable=no-member
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the DWD-Weather-Warnings."""
+
+        data = {}
+        data['region_name'] = self._api.region_name
+
+        if self._api.region_id is not None:
+            data['region_id'] = self._api.region_id
+
+        if self._api.region_state is not None:
+            data['region_state'] = self._api.region_state
+
+        if self._api.data['time'] is not None:
+            data['last_update'] = dt_util.as_local(
+                dt_util.utc_from_timestamp(self._api.data['time'] / 1000))
+
+        if self._var_id == 'current_warning_level':
+            prefix = 'current'
+        elif self._var_id == 'advance_warning_level':
+            prefix = 'advance'
+
+        data['warning_count'] = self._api.data[prefix + '_warning_count']
```

Comment:
Local variable 'prefix' might be referenced before assignment. Compare L122, L124.

Reply:
ok I will add a else path and throw an exception there...